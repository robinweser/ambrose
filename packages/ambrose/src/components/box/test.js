import React from 'react'

import Box from '.'

import createSnapshot from '../../testing/createSnapshot'

describe('<Box />', () => {
  it('renders correctly with default props', () => {
    const snapshot = createSnapshot(<Box>Children</Box>)
    expect(snapshot).toMatchSnapshot()
  })
  it('renders correctly with backgroundColor', () => {
    const snapshot = createSnapshot(
      <Box backgroundColor="lightgrey">Children</Box>
    )
    expect(snapshot).toMatchSnapshot()
  })
  it('renders correctly with paddingHorizontal', () => {
    const snapshot = createSnapshot(<Box paddingHorizontal={10}>Children</Box>)
    expect(snapshot).toMatchSnapshot()
  })
  it('renders correctly with paddingVertical', () => {
    const snapshot = createSnapshot(<Box paddingVertical={10}>Children</Box>)
    expect(snapshot).toMatchSnapshot()
  })
})
