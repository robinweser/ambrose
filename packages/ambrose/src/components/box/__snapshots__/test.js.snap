// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> renders correctly with backgroundColor 1`] = `
".cls_0 {
  display: contents;
}
.cls_1 {
  box-sizing: border-box;
  flex-direction: column;
  flex-wrap: nowrap;
  flex-grow: 0;
  flex-shrink: 0;
  flex-basis: auto;
  align-items: stretch;
  display: flex;
  background-color: lightgrey;
}


<div className=test-theme cls_0>
  <div className=cls_1>Children</div>
</div>;
"
`;

exports[`<Box /> renders correctly with default props 1`] = `
".cls_0 {
  display: contents;
}
.cls_1 {
  box-sizing: border-box;
  flex-direction: column;
  flex-wrap: nowrap;
  flex-grow: 0;
  flex-shrink: 0;
  flex-basis: auto;
  align-items: stretch;
  display: flex;
}


<div className=test-theme cls_0>
  <div className=cls_1>Children</div>
</div>;
"
`;

exports[`<Box /> renders correctly with paddingHorizontal 1`] = `
".cls_0 {
  display: contents;
}
.cls_1 {
  box-sizing: border-box;
  flex-direction: column;
  flex-wrap: nowrap;
  flex-grow: 0;
  flex-shrink: 0;
  flex-basis: auto;
  align-items: stretch;
  display: flex;
  padding-left: 40px;
  padding-right: 40px;
}


<div className=test-theme cls_0>
  <div className=cls_1>Children</div>
</div>;
"
`;

exports[`<Box /> renders correctly with paddingVertical 1`] = `
".cls_0 {
  display: contents;
}
.cls_1 {
  box-sizing: border-box;
  flex-direction: column;
  flex-wrap: nowrap;
  flex-grow: 0;
  flex-shrink: 0;
  flex-basis: auto;
  align-items: stretch;
  display: flex;
  padding-bottom: 40px;
  padding-top: 40px;
}


<div className=test-theme cls_0>
  <div className=cls_1>Children</div>
</div>;
"
`;
